OCPVE-337: Add an architecture diagram (#341)
openshift/lvm-operator: README.md

The LVM Operator, in conjunction with the TopoLVM CSI Driver, Volume Group Manager, and other related components, collectively comprise the Logical Volume Manager Storage (LVMS) solution.

> Here is a brief overview of how the operator works. See [here](docs/design/architecture.md) for the architecture diagram.

Reviewer: ```suggestion
Here is a brief overview of how the Operator works. See [here](docs/design/architecture.md) for the architecture diagram.
```

## Metrics

> The LVM Operator runs a metrics exporter sidecar to export Prometheus metrics. To enable monitoring on OpenShift clusters, you must assign the label `openshift.io/cluster-monitoring` to the namespace in which LVMS is running.

Reviewer: ```suggestion
The LVM Operator runs a metrics exporter sidecar to export Prometheus metrics. To enable monitoring on OpenShift clusters, assign the `openshift.io/cluster-monitoring` label to the same namespace that you deployed LVMS to.
```
